kahypar/io/hypergraph_io.h

```diff
@@ -77,22 +146,22 @@ static inline void readHypergraphFile(const std::string& filename, HypernodeID&
                                        hypergraph_type == HypergraphType::EdgeAndNodeWeights ?
                                        true : false;
 
+    std::vector<size_t> line_number_vector;
     index_vector.reserve(static_cast<size_t>(num_hyperedges) +  /*sentinel*/ 1);
     index_vector.push_back(edge_vector.size());
+    if (line_number_vector != nullptr) {
```

Comment:
Conceptionally, yes. However, the way the function is currently written it only takes the `line_number_vector` parameter, which is null if no additional input validation occurs after reading the file and is otherwise used as an output parameter for mapping vertices/edges to input lines. That seemed simpler to me than adding even more parameters